v0/destinations/google_adwords_offline_conversions/transform.js

```diff
@@ -0,0 +1,280 @@
+const sha256 = require("sha256");
+const { get, set, cloneDeep } = require("lodash");
+const moment = require("moment");
+const { EventType } = require("../../../constants");
+const {
+  getHashFromArrayWithDuplicate,
+  constructPayload,
+  removeHyphens,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  simpleProcessRouterDest,
+  getHashFromArray,
+  getFieldValueFromMessage
+} = require("../../util");
+const ErrorBuilder = require("../../util/error");
+const {
+  CLICK_CONVERSION,
+  CALL_CONVERSION,
+  trackClickConversionsMapping,
+  trackCallConversionsMapping
+} = require("./config");
+const {
+  validateDestinationConfig,
+  getAccessToken,
+  removeHashToSha256TypeFromMappingJson
+} = require("./utils");
+
+/**
+ * get conversions depending on the type set from dashboard
+ * i.e click conversions or call conversions
+ * @param {*} message
+ * @param {*} metadata
+ * @param {*} param2
+ * @param {*} event
+ * @param {*} conversionType
+ * @returns
+ */
+const getConversions = (
+  message,
+  metadata,
+  { Config },
+  event,
+  conversionType
+) => {
+  let payload;
+  let endpoint;
+  const filteredCustomerId = removeHyphens(Config.customerId);
+  const { hashUserIdentifier } = Config;
+  const { properties } = message;
+
+  if (conversionType === "click") {
+    // click conversions
+    let updatedClickMapping = cloneDeep(trackClickConversionsMapping);
+
+    if (hashUserIdentifier === false) {
+      updatedClickMapping = removeHashToSha256TypeFromMappingJson(
+        updatedClickMapping
+      );
+    }
+
+    payload = constructPayload(message, updatedClickMapping);
+
+    // either of email or phone should be passed
+    // defaultUserIdentifier depends on the webapp configuration
+    // Ref - https://developers.google.com/google-ads/api/rest/reference/rest/v11/customers/uploadClickConversions#ClickConversion
+
+    if (Config.defaultUserIdentifier === "email") {
+      let email = getFieldValueFromMessage(message, "email");
+      if (email) {
+        email = Config.hashUserIdentifier ? sha256(email).toString() : email;
+        set(payload, "conversions[0].userIdentifiers[0].hashedEmail", email);
+      }
```

Comment:
What if email or phone is not available? Are they accepting the call without any user identifier?

Reply:
I think either is required